lib/spack/spack/solver/concretize.lp

```diff
@@ -77,15 +81,21 @@ depends_on(Package, Dependency) :- depends_on(Package, Dependency, _).
 dependency_holds(Package, Dependency, Type) :-
   dependency_condition(ID, Package, Dependency),
   dependency_type(ID, Type),
-  condition_holds(ID).
+  condition_holds(ID),
+  not external(Package).
+
+% We cut off dependencies of externals (as we don't really know them).
+% Don't impose constraints on dependencies that don't exist.
```

Comment:
Is one of the two lines redundant? (They are both correct, but they seem to say the same thing to me)

Reply:
IMO they're not redundant -- one's about how spack works and the other's about what we have to do in the solve :)